### PR TITLE
Fixup tests

### PR DESCRIPTION
**Pull request by stefanpenner**
- make `ember test` work
- make `ember test --server` work
- ensure existing `npm test` and its various scenarios continue to work
- upgrade testem
- ensure our various test permutations work with phantomjs 2

### CODE REVIEW

**Comment by rwjblue** [comment]
I think that https://github.com/emberjs/ember.js/blob/master/bin/run-sauce-tests.js#L36-L51 needs tweaking (or can be removed?) to fix the sauce labs tests.

**Comment by stefanpenner** [comment]
Ya. I'll take a look in about 30 min

**Comment by stefanpenner** [comment]
@rwjblue we are green, and now also tests FF.

File: packages/ember-htmlbars/tests/integration/select_in_template_test.js [hunk 1]
```diff
@@ -152,35 +152,39 @@ QUnit.test('works from a template', function() {
...
-
-  view = EmberView.create({
-    user: userOne,
-    selectView: SelectView,
-    template: compile(
-      '{{view view.selectView viewName="select"' +
-      '    content=view.user.options' +
-      '    selection=view.user.selectedOption}}'
-    )
+if (window.navigator.userAgent.indexOf('Firefox') === -1) {
```
Comment (rwjblue) [inline packages/ember-htmlbars/tests/integration/select_in_template_test.js]: Can you use the feature detect that exists in the `ember-metal/environment` module [here](https://github.com/emberjs/ember.js/blob/master/packages/ember-metal/lib/environment.js#L28) instead?

Comment (rwjblue) [inline packages/ember-htmlbars/tests/integration/select_in_template_test.js]: Specifically, you should be able to do:

```javascript
import environment from 'ember-metal/environment';

if (environment.isFirefox) {
 /* test here */
}
```

File: packages/ember-htmlbars/tests/integration/select_in_template_test.js [hunk 2]
```diff
@@ -152,35 +153,39 @@ QUnit.test('works from a template', function() {
...
-
-  view = EmberView.create({
-    user: userOne,
-    selectView: SelectView,
-    template: compile(
-      '{{view view.selectView viewName="select"' +
-      '    content=view.user.options' +
-      '    selection=view.user.selectedOption}}'
-    )
+if (environment.isFireFox) {
```
Comment (rwjblue) [inline packages/ember-htmlbars/tests/integration/select_in_template_test.js]: I think you mean `if (!environment.isFirefox)`

**Comment by stefanpenner** [comment]
Will fix shortly

**Comment by stefanpenner** [comment]
sorry, been using sauce as my tester :P.  We may be fixed now.

**Comment by stefanpenner** [comment]
maybe it work better if i pushed my work....

**Pull request merged**